src/hooks/use-dnd-sensor.ts

```diff
@@ -0,0 +1,59 @@
+import { useState, useCallback, useMemo } from 'react'
+import {
+  useSensor,
+  useSensors,
+  PointerSensor,
+  TouchSensor,
+  DragStartEvent,
+  DragEndEvent,
+  Active
+} from '@dnd-kit/core'
+import { arrayMove } from '@dnd-kit/sortable'
+
+type IdProp = 'id' | '_id'
+
+const useDndSensor = <T extends { [K in IdProp]?: string }>(
```

Comment:
I used this pattern. What do you think about it?

t```
ype IdType1 = { id: string | number; _id?: string | number }
type IdType2 = { id?: string | number; _id: string | number }

type IdType = IdType1 | IdType2

interface UseDndSensorProps<T extends IdType> {
  items: T[]
  setItems: (items: T[]) => void
  idProp: 'id' | '_id'
}

const useDndSensor = <T extends IdType>({
```

Reply:
I think it is a pretty good solution here because as you had shown currently we have items, that for some reason pass down both `_id` and `id` prop, which most likely incorrect, but this is not related to this issue.

I also suggest renaming from `IdType` to `ObjectWithId` to be more precise in naming. Then there is no need to create types `IdType1` and `IdType2`, just use them inline inside `ObjectWithId`

Reply:
Changed